### ISSUE TITLE
Khalida's defeat trait and a few missions referring to missile units missing a few TK units

### ISSUE THREAD

**Issue by allenhu0912** [chadvandy/cbfm_wh3#1403]
![20240121234157_1](https://github.com/chadvandy/cbfm_wh3/assets/54481962/53a01638-f262-4430-a63d-e3a1001f5c30)

**Comment by Grimmys** [comment]
Ok, so for this one, I think it's working this way: 

In `db/effect_bonus_value_ids_unit_sets_tables/data__`, it is specified for this trait effect (`wh_main_effect_force_army_campaign_experience_base_ranged_units`), all units matching `missile_units` unit set are affected.

Then, in `db/unit_set_to_unit_junctions_tables/data__`, it is specified that units having caste `missile_cavalry`, `missile_infantry` and `warmachine` are considered as `missile_units`. There is also an extra entry in this table for specifying that Skeleton Archer Chariot is also considered as `missile_units`, despite not having a caste matching the category.

It is explaining why Ushabti Great Bow is not affected, because its caste is `monstruous_infantry` (according to `db/main_units_table/data__`).

For the same reason as for #1411, Casket of Souls is also not affected by the effect (despite Skull Catapult being), because its caste is `chariot`.

I would like the opinion of experienced people on this one, because I feel like if we consider Ushabti Great as valid target, then we could also consider Bone Giant, and kinda Sepulchral as well etc.
So were is the limit?

**TL;DR for people not interested by technical side:** fix would be easy, but the question is for which other units it would make sense as well.
For Casket of Souls, maybe we should really change its caste to fix multiple issues, I'm not sure it's right to consider it differently than Skull Catapult.

**Comment by Harmandodelavega** [comment]
Again something they fixed in WH2 got lost in the porting

**Comment by Harmandodelavega** [comment]
Add it to everything

**Comment by Harmandodelavega** [comment]
Btw this is a defeat trait so it will affect every race, but it's a little much to ask to go through everything so we'll keep it short

**Comment by Grimmys** [comment]
Yeaaah, recruit rank upgrade for everybody :fire:

What I will fix is the effect itself btw (not the trait), so it should also fix the "Missile Focus" bonus got after hiring a missile unit and maybe also the quest itself asking you to hire missile type unit.

For other races, I guess we could fix them on the fly when someone will complain about an issue.

**Comment by Harmandodelavega** [comment]
It's a tongue twister, if anyone comes up with a better title before me go ahead
